deltacat/compute/compactor/steps/materialize.py

```diff
@@ -41,6 +41,54 @@ def materialize(
         schema: Optional[pa.Schema] = None,
         deltacat_storage=unimplemented_deltacat_storage) -> MaterializeResult:
 
+    def _materialize(
+            compacted_tables: List[pa.Table],
+            compacted_tables_record_count: int) -> MaterializeResult:
+        compacted_tables_size = sum([TABLE_CLASS_TO_SIZE_FUNC[type(tbl)](tbl)
+                                     for tbl in compacted_tables])
+        logger.debug(f"Uploading {len(compacted_tables)} compacted tables "
+                     f"with size: {compacted_tables_size} bytes "
+                     f"and record count: {compacted_tables_record_count}")
+        compacted_table = pa.concat_tables(compacted_tables)
+        if compacted_file_content_type in DELIMITED_TEXT_CONTENT_TYPES:
+            # convert to pandas since pyarrow doesn't support custom delimiters
+            # and doesn't support utf-8 conversion of all types (e.g. Decimal128)
```

Comment:
latest version of pyarrow now supports utf-8 conversion of decimal.

Reply:
Maybe a fresh TODO here would be a good idea, since changing this would be a bit out-of-scope.

Reply:
Ok.